modules/memfault-firmware-sdk/memfault_location_metrics.c

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/sys/printk.h>
+#include <zephyr/logging/log.h>
+#include <modem/location.h>
+
+#include "memfault/metrics/metrics.h"
+#include "memfault_location_metrics.h"
+
+LOG_MODULE_DECLARE(memfault_ncs_metrics, CONFIG_MEMFAULT_NCS_LOG_LEVEL);
+
+static bool session_in_progress = false;
+
+static void memfault_location_method_to_string(char *buf, size_t buf_len,
+					       enum location_method method)
+{
+	switch (method) {
+	case LOCATION_METHOD_CELLULAR:
+		snprintk(buf, buf_len, "Cellular");
```

Comment:
Nit, just return `const char *` to the `"Cellular"` etc strings, instead of copying. Use `%s` in the calling function.